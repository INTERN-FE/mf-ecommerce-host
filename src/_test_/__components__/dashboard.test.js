import { MemoryRouter } from "react-router-dom";
import { shallow, configure } from "enzyme";
import Adapter from "@cfaester/enzyme-adapter-react-18";
import MainPage from "../../pages/main-page";

configure({ adapter: new Adapter() });

jest.mock(
  "mf_list_product/ListProduct",
  () => ({
    ListProduct: () => "ListProduct",
  }),
  { virtual: true }
);

jest.mock(
  "mf_list_product/DetailProduct",
  () => ({
    DetailProduct: () => "DetailProduct",
  }),
  { virtual: true }
);

describe("eCommerce Detail Product", () => {
  it("It should render Detail Product", () => {
    const url = ["/detail/1"];
    const components = (
      <MemoryRouter initialEntries={url}>
        <MainPage />
      </MemoryRouter>
    );

    const wrapper = shallow(components);
    expect(wrapper).toMatchSnapshot();
  });
});
