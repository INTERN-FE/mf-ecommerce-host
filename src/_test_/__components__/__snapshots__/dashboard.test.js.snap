// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eCommerce Detail Product It should render Detail Product 1`] = `ShallowWrapper {}`;
